### PR TITLE
chore(ux): improve error message for `oras tag list`

### LINKED ISSUE #1210: improve error message when user mistakenly uses `tag list` command
Current behavior and output:

```console
$ oras tag list ghcr.io/oras-project/oras
Error: unable to add tag for 'list': invalid reference: missing repository
```

Suggested error message:

```console
$ oras tag list ghcr.io/oras-project/oras
Error: There is no "list" sub-command for "oras tag" command
Usage: oras tag [flags] <name>{:<tag>|@<digest>} <new_tag> [...]
If you want to list available tags in a repository, use "oras repo tags"

### PR DESCRIPTION
**Pull request by qweeah**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1210 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/root/tag.go [hunk 1]
```diff
@@ -58,7 +59,20 @@ Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.1', 'v1.0
...
+				if opts.IsOCILayout {
+					container = "an OCI image layout"
+				}
+				return &oerrors.Error{
+					Err:            errors.New(`there is no "list" sub-command for "oras tag" command`),
+					Usage:          fmt.Sprintf("%s %s", cmd.CommandPath(), cmd.Use),
+					Recommendation: fmt.Sprintf(`If you want to list available tags in %s, use "oras repo tags"`, container),
+				}
+			}
+			return oerrors.CheckArgs(argument.AtLeast(1), "the to-be-retage artifact and the tags to be added")(cmd, args)
```
Comment (TerryHowe) [inline cmd/oras/root/tag.go:74]:
```suggestion
			return oerrors.CheckArgs(argument.AtLeast(1), "the to-be-retagged artifact and the tags to be added")(cmd, args)
```

Comment (shizhMSFT) [reply cmd/oras/root/tag.go:74]: nice catch

Comment (qweeah) [reply cmd/oras/root/tag.go]: Did a little rephrasing.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**